src/twisted/application/runner/test/test_runner.py

```diff
@@ -68,11 +68,31 @@ def test_runInOrder(self):
         """
         L{Runner.run} calls the expected methods in order.
         """
-        runner = DummyRunner({})
+
+        calledMethods = []
+
+        self.patch(
```

Comment:
Ideally neither subclassing or `.patch` would be required, but subclassing is (in this case, anyway) I'd say tolerable.

`.patch` is the worst possible way to change things to make them testable, because they're process-global.  It should always be at the bottom of the mocking hierarchy.  It is a good "option of last resort" for garbage legacy code that is really not indirectable in any other way, and hence isn't something we should remove from _trial_, but we should avoid it in our own tests as much as possible.  Hopefully @moshez will concur on this general hierarchy: composition > inheritance > patching.

Moving on to a practical problems involved with this case: let's say that `trial` wants to adopt `Runner` as a way to run itself.  If there's any problem with the patching, the global instance that Trial is using for its _own_ task-running will suddenly have these methods replaced.  Now, in this *exact* case probably nothing happens, but in the rare case that some horrible bug causes a patch cleanup to fail, your process can no longer shut down, or finish starting up, because this test's patches are still in place.